File: src/main/java/utils/RandomNumberGenerator.java
package utils;

import java.util.Random;

public class RandomNumberGenerator implements NumberGenerator {

    public static final int RANDOM_UPPER_BOUND = 10;
    private static final Random RANDOM = new Random();

    @Override
    public int generate() {
        return RANDOM.nextInt(RANDOM_UPPER_BOUND);
    }
}

File: src/main/java/utils/RandomNumberGenerator.kt
package utils

import java.util.*

class RandomNumberGenerator : NumberGenerator {

    override fun generate(): Int {
        return RANDOM.nextInt(RANDOM_UPPER_BOUND)
    }

    companion object {
        private const val RANDOM_UPPER_BOUND = 10
        private val RANDOM = Random()
    }
}
